packages/components/socialFeed/Map/Map.web.tsx

```diff
@@ -0,0 +1,254 @@
+  TileLayer,
+  Marker,
+  Popup,
+  useMapEvents,
+  useMap,
+} from "react-leaflet";
+import MarkerClusterGroup from "react-leaflet-cluster";
+import { HeatmapLayer } from "react-leaflet-heatmap-layer-v3/lib";
+import { View } from "react-native";
+
+import { Post } from "@/api/feed/v1/feed";
+import { MapProps } from "@/components/socialFeed/Map/Map.types";
+import { ArticleMapPost } from "@/components/socialFeed/Map/MapPosts/ArticleMapPost";
+import { MusicMapPost } from "@/components/socialFeed/Map/MapPosts/MusicMapPost";
+import { NormalMapPost } from "@/components/socialFeed/Map/MapPosts/NormalMapPost";
+import { PictureMapPost } from "@/components/socialFeed/Map/MapPosts/PictureMapPost";
+import { VideoMapPost } from "@/components/socialFeed/Map/MapPosts/VideoMapPost";
+import { useFetchFeedLocation } from "@/hooks/feed/useFetchFeed";
+import { useSelectedNetworkId } from "@/hooks/useSelectedNetwork";
+import {
+  DEFAULT_MAP_POSITION,
+  getMapPostIconColorRgba,
+  getMapPostIconSVGString,
+  MAP_LAYER_URL,
+} from "@/utils/feed/map";
+import { zodTryParseJSON } from "@/utils/sanitize";
+import {
+  CustomLatLngExpression,
+  PostCategory,
+  ZodSocialFeedPostMetadata,
+} from "@/utils/types/feed";
+
+interface MarkerPopup {
+  position: CustomLatLngExpression;
+  post: Post;
+  fileURL?: string;
+}
+
+// Custom map post icon
+const postIcon = (postCategory: PostCategory) => {
+  const size = 32;
+  const borderWidth = 1;
+  const sizeWithBorders = 32 + borderWidth * 2;
+  return new DivIcon({
+    html: `<div style="border-radius: 99px;
+    height: ${size}px; width: ${size}px; border: 1px solid #A3A3A3;
+     background-color: rgba(${getMapPostIconColorRgba(postCategory)}); display: flex; align-items: center; justify-content: center;">${getMapPostIconSVGString(postCategory)}</div>`,
+    className: "",
+    iconSize: [sizeWithBorders, sizeWithBorders],
+  });
+};
+
+// Custom cluster icon
+const clusterIcon = function (cluster: any) {
+  return new DivIcon({
+    html: `<div class="cluster-icon-wrapper"><span class="cluster-icon">${cluster.getChildCount()}</span></div>`,
+    className: "custom-marker-cluster",
+    iconSize: point(33, 33, true) as PointExpression,
+  });
+};
+
+export const Map: FC<MapProps> = ({
+  locationToCenter = DEFAULT_MAP_POSITION,
+  style,
+  creatingPostLocation,
+  creatingPostCategory = -1,
+}) => {
+  const selectedNetworkId = useSelectedNetworkId();
+  const [bounds, setBounds] = useState<LatLngBounds | null>(null);
+  const [isMapLoaded, setMapLoaded] = useState(false);
+  // Prevent infinite rendering after locationSelected update
+  const [localLocationSelected, setLocalLocationSelected] = useState<
+    CustomLatLngExpression | undefined
+  >(creatingPostLocation);
+  useEffect(() => {
+    setLocalLocationSelected(creatingPostLocation);
+  }, [creatingPostLocation]);
+
+  // ---- Fetch existing posts that have a location and display them as markers
+  const { data } = useFetchFeedLocation({
```

Comment:
just verify to ensure that data.list is unchanged to ensure that will not trigger useMemo(..., [posts]) every circle

Reply:
I ensured `useFetchFeedLocation` and so ` useMemo(..., [posts])` is not triggered too much.
It's triggered only depending on `UpdateBoundsOnMapLoad` and `UpdateBoundsOnMapEvents`, so :
- When the map is shown (once)
- When the user zoom/dezoom or move the map